algebra/tests/matrix/basic_interactions.c

```diff
@@ -292,3 +292,49 @@ TEST_GROUP_RUNNER(group_matrix_bufAlloc)
 	RUN_TEST_CASE(group_matrix_bufAlloc, matrix_bufAlloc_readAndWrite);
 	RUN_TEST_CASE(group_matrix_bufAlloc, matrix_bufAlloc_invalidArgs);
 }
+
+
+/* ##############################################################################
+ * ---------------------        matrix_bufFree tests       ----------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_matrix_bufFree);
+
+
+TEST_SETUP(group_matrix_bufFree)
+{
+	dynMat.data = NULL;
+}
+
+
+TEST_TEAR_DOWN(group_matrix_bufFree)
+{
+	free(dynMat.data);
+}
+
+
+TEST(group_matrix_bufFree, matrix_bufFree_passingDataEqualNULL)
+{
+	/* dynMat.data is equal to NULL - see TEST_SETUP */
+	matrix_bufFree(&dynMat);
+
+	TEST_ASSERT_NULL(dynMat.data);
+}
```

Comment:
```suggestion
TEST(group_matrix_bufFree, matrix_bufFree_doubleFreeSafe)
{
	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK, matrix_bufAlloc(&dynMat, ROWS, COLS))
	
	matrix_bufFree(&dynMat);
	TEST_ASSERT_NULL(dynMat.data);
	
	matrix_bufFree(&dynMat);
	TEST_ASSERT_NULL(dynMat.data);
}
```
I think this way this test covers more scenarios and is more understandable. (works only after rebase)